packages/components/src/Actions/ActionDropdown/ActionDropdown.component.js

```diff
@@ -108,67 +108,100 @@ function getMenuItem(item, index, getComponent) {
 };
  <ActionDropdown {...props} />
  */
-function ActionDropdown(props) {
-	const {
-		bsStyle,
-		hideLabel,
-		icon,
-		items,
-		label,
-		link,
-		onSelect,
-		tooltipPlacement,
-		tooltipLabel,
-		getComponent,
-		components,
-		className,
-		...rest
-	} = props;
-
-	const Renderers = Inject.getAll(getComponent, { MenuItem, DropdownButton });
-	const injected = Inject.all(getComponent, components, InjectDropdownMenuItem);
-	const title = (
-		<span className="tc-dropdown-button-title">
-			{icon ? <Icon name={icon} /> : null}
-			{hideLabel ? null : <span className="tc-dropdown-button-title-label">{label}</span>}
-		</span>
-	);
-	const style = link ? 'link' : bsStyle;
+class ActionDropdown extends React.Component {
+	constructor(props) {
+		super(props);
+		this.onToggle = this.onToggle.bind(this);
+		this.state = {
+			dropup: false,
+		};
+	}
+
+	onToggle(isOpen, event) {
+		if (!isOpen) {
+			return;
+		}
+		const dropdownTrigger = event.target;
+		const dropdownMenu = event.target.nextSibling;
+		let dropdownContainer = dropdownTrigger.parentElement;
+		while (dropdownContainer && !dropdownContainer.classList.contains('tc-dropdown-container')) {
```

Comment:
it is specific for the list but if i want use the ActionDropdown in a other context, i can't use it

Reply:
for example in the ActionDropDown, if i want use this feature to avoid to collapse below the viewport, i have to previously add a class tc-dropdown-container on the element which wraps my ActionDropdown. It is not automatically :

So, if it is this case, can you add an example on this story: http://1403.talend.surge.sh/components/?selectedKind=ActionDropdown&selectedStory=default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel


![image](https://user-images.githubusercontent.com/20772261/40988577-c04177f2-68eb-11e8-9d05-3410c51ad4b7.png)